### ISSUE TITLE
bad propfind since 2.4.3

### ISSUE THREAD

**Issue by srguiwiz** [OpenMarshal/npm-WebDAV-Server#50]
Had code using 2.4.1.  Testing from client code in Chrome, Firefox fetch propfind depth 1.  Ran fine for three months.  Also access from Windows Explorer and Mac OS X Finder.

Went to 2.4.3.  Broke.  See stuff like

```
	<D:response>
		<D:href>https://www.example.com/whatever.html</D:href>
		<D:propstat>
			<D:status>HTTP/1.1 404 Not Found</D:status>
		</D:propstat>
	</D:response>
```

Went one version back to 2.4.2.  Still works.

The failure was definitely observed in Chrome and Mac OS X Finder at least.

I hope further testing on your side can reproduce this problem.

**Comment by AdrienCastex** [comment]
I published the new version on npm `v2.4.4`.

I added automated tests to verify the digest authentication. They are all green now.

I let you close the issue if everything is working well for you.

**Comment by AdrienCastex** [comment]
Hello,

I don't reproduce the problem.

Can you give me a request which produces such error and the settings of the server (hide the sensitive information) ?

What do you mean by Chrome ? Correct me if I am wrong but Chrome itself doesn't handle webdav.

**Comment by srguiwiz** [comment]
My request body is:
```
<?xml version="1.0" encoding="utf-8"?>
<D:propfind xmlns:D='DAV:'>
	<D:prop>
		<D:displayname/>
		<D:resourcetype/>
		<D:getcontenttype/>
		<D:getlastmodified/>
	</D:prop>
</D:propfind>
```
And relevant request headers are:
```
Depth: 1
Content-Type: application/xml; charset='utf-8'
```
That one invocation is with a fetch in JavaScript in Chrome.  Nothing changes except upgrade of this library.  Before it returned good XML.  Now it returns essentially a list of 404 for every file:
```
<D:multistatus xmlns:D="DAV:">
	<D:response>
		<D:href>http://server/directory1/directory1a/</D:href>
		<D:propstat>
			<D:status>HTTP/1.1 200 OK</D:status>
			<D:prop>
				<D:getlastmodified>Sun, 20 Jan 2019 17:38:20 GMT</D:getlastmodified>
				<D:resourcetype>
					<D:collection/>
				</D:resourcetype>
				<D:displayname>directory1a</D:displayname>
			</D:prop>
		</D:propstat>
		<D:propstat>
			<D:prop>
				<a:getcontenttype xmlns:a="DAV:"/>
			</D:prop>
			<!-- this is okay, a directory has no content type -->
			<D:status>HTTP/1.1 404 Not Found</D:status>
		</D:propstat>
	</D:response>
	<D:response>
		<D:href>http://server/example1.html</D:href>
		<D:propstat>
			<D:status>HTTP/1.1 404 Not Found</D:status>
		</D:propstat>
	</D:response>
	<D:response>
		<D:href>http://server/example2.html</D:href>
		<D:propstat>
			<D:status>HTTP/1.1 404 Not Found</D:status>
		</D:propstat>
	</D:response>
```
And there is the problem:  Inside directory `http://server/directory1/directory1a/` the files would have to be `http://server/directory1/directory1a/example1.html` but it is looking for `http://server/example1.html`.

**Comment by AdrienCastex** [comment]
Hello,

I think I found the problem. The path used for the children of a resource was the one of the file system (relative to the file system root), not the server path ("absolute" path, from the root of the server).

I think I fixed it (I added tests to check this case), it will be available on the `v2.4.5`.

Let me know if it doesn't fix the problem for you.

**Comment by srguiwiz** [comment]
Looks good, appears to be working since 2.4.5.  Thank you!
